pkg/crawler/crawler.go

```diff
@@ -101,7 +105,13 @@ func (c *Crawler) Crawl(ctx context.Context) error {
 				defer c.limit.Release(1)
 				defer c.wg.Done()
 				if err := c.Visit(ctx, url); err != nil {
-					errCh <- xerrors.Errorf("visit error: %w", err)
+					// There might be a case where we get 2 errors at the same time.
+					// In this case we close `errCh` after reading the first error
+					// and get panic for the second error
+					// That's why we need to return the error once.
+					c.errOnce.Do(func() {
+						errCh <- xerrors.Errorf("visit error: %w", err)
+					})
```

Comment:
Since we already have a context for cancellation, can't we check the context?

```suggestion
					select {
					case <-ctx.Done():
						return
					default:	
						errCh <- xerrors.Errorf("visit error: %w", err)
					})
```

Reply:
hm... i will check this.

Reply:
If we use a buffered channel, it works, but I don't know why it works...
https://go.dev/play/p/0un699UN6-P

```
	errCh := make(chan error, 1)
```

Reply:
If i understood correctly - problem with `default`.
Your solution works with non-buffered channel - https://go.dev/play/p/SN26pwfjMqR

Reply:
Interesting. If it's a non-buffered channel, `errCh` receives an error without blocking. So, I didn't think it helped. I must be missing something.
Alternatively, writing it in `select` may not cause panic because it will make sure that the channel is open before sending it.

Reply:
https://go.dev/tour/concurrency/5 says:
`A select blocks until one of its cases can run, then it executes that case. It chooses one at random if multiple are ready. `

https://go.dev/tour/concurrency/6 says:
`The default case in a select is run if no other case is ready. `

UUIC closed channel is not case that can be run.

Reply:
Yes, but I thought that `errCh` is always considered ready if it is an unbuffered channel.

Reply:
It seems filled unbuffered channel is not `ready` for `select`.